server/topic.go

```diff
@@ -1131,8 +1148,8 @@ func (t *Topic) thisUserSub(h *Hub, sess *Session, pkt *ClientComMessage, asUid
 
 	var changed bool
 
-	isChanSub := isChannel(pkt.Original)
-	if isChanSub && !t.isChan {
+	isChanSub, err := t.verifyChannelAccess(pkt.Original)
+	if err != nil {
 		// User should not be able to address non-channel topic as channel.
 		sess.queueOut(ErrNotFoundReply(pkt, now))
 		return changed, types.ErrNotFound
```

Comment:
Should we return `err` here instead of `ErrNotFound`?

Reply:
The `err` is `types.ErrNotFound` in this case. Returning an explicit error is easier to read.